### DISCUSSION TITLE
How to login with 2FA e-mail codes?

### DISCUSSION THREAD

**Discussion by Apreche** [MarshalX/atproto#455] [Questions] [answered]
My account has 2FA enabled. That means every login I get e-mailed a code that I have to use to sign in.
If I try to login using atproto `client.login()` it throws an `UnauthorizedError` exception and I receive an e-mail with the code.
I can't see anywhere in the documentation or the source code of this API client library any way to use that e-mail code to successfully login. Is there something I’m missing or is it unsupported?

Bluesky currently does not support any modern authentication methods like FIDO, passkey, etc. The only option I see in the settings is to make app passwords. I tried that, and it worked. Is that the current best practice for this scenario?

**Comment by MarshalX** [comment, marked as answer]
Hey, yes. Using app password is the most right approach for now. The near future is probably oAuth
